Backpack/src/main/java/net/skyscanner/backpack/bottomnav/BpkBottomNav.kt

```diff
@@ -0,0 +1,122 @@
+    background = ContextCompat.getDrawable(context, R.drawable.bpk_bottom_nav_background)
+    minimumHeight = resources.getDimensionPixelSize(R.dimen.bpk_bottom_nav_height)
+    itemTextColor = ContextCompat.getColorStateList(context, R.color.bpk_bottom_nav_selector)
+    itemIconTintList = itemTextColor
+    ViewCompat.setElevation(this, resources.getDimension(R.dimen.bpkElevationLg))
+  }
+
+  fun addItem(id: Int, title: String, icon: Drawable): MenuItem =
+    menu.add(Menu.NONE,
+      id,
+      menu.size(),
+      SpannableStringBuilder().append(title, fontSpan, Spannable.SPAN_INCLUSIVE_INCLUSIVE)
+    )
+      .setIcon(icon)
+
+  fun addItem(id: Int, @StringRes title: Int, @DrawableRes icon: Int): MenuItem =
+    addItem(id, resources.getString(title), ContextCompat.getDrawable(context, icon)!!)
+
+  fun addOnNavigationItemSelectedListener(listener: (MenuItem, Int) -> Unit) {
+    listeners.selected += listener
+  }
+
+  fun addOnNavigationItemReselectedListener(listener: (MenuItem, Int) -> Unit) {
+    listeners.reselected += listener
+  }
+
+  fun removeOnNavigationItemSelectedListener(listener: (MenuItem, Int) -> Unit) {
+    listeners.selected -= listener
+  }
+
+  fun removeOnNavigationItemReselectedListener(listener: (MenuItem, Int) -> Unit) {
+    listeners.reselected -= listener
+  }
+
+  @Deprecated("Use add/remove OnNavigationItemSelectedListener instead",
+    replaceWith = ReplaceWith("addOnNavigationItemSelectedListener"))
+  override fun setOnNavigationItemSelectedListener(listener: OnNavigationItemSelectedListener?) {
+    TODO("Not supported")
+  }
+
+  @Deprecated("Use add/remove OnNavigationItemSelectedListener instead",
+    replaceWith = ReplaceWith("addOnNavigationItemReselectedListener"))
+  override fun setOnNavigationItemReselectedListener(listener: OnNavigationItemReselectedListener?) {
+    TODO("Not supported")
+  }
+
+  private class ListenersDelegate(private val menu: Menu) : OnNavigationItemSelectedListener, OnNavigationItemReselectedListener {
+
+    val reselected = mutableListOf<(MenuItem, Int) -> Unit>()
+    val selected = mutableListOf<(MenuItem, Int) -> Unit>()
+
+    override fun onNavigationItemReselected(item: MenuItem) {
+      val index = findIndexOf(item)
+      if (index != -1) {
+        reselected.forEach { it.invoke(item, index) }
+      }
+    }
+
+    override fun onNavigationItemSelected(item: MenuItem): Boolean {
+      val index = findIndexOf(item)
+      if (index != -1) {
+        selected.forEach { it.invoke(item, index) }
+      }
+      return true
+    }
+
+    private fun findIndexOf(item: MenuItem): Int {
+      for (i in 0 until menu.size()) {
+        if (menu.getItem(i) == item) {
+          return i
+        }
+      }
+      return -1
+    }
+  }
+
+  // We have to create a custom span as a BpkFontSpan causes some animation glitches as it sets text size as well.
+  private class BottomNavSpan(private val font: BpkText.FontDefinition) : CharacterStyle() {
```

Comment:
Does that mean we are not using a valid Backpack text size?

Reply:
You cannot customise the text size of the Material Bottom Nav -- but as far as I can see it uses one of our sizes.